### PR TITLE
Add link for creating account for anonymous users

### LINKED ISSUE #218: Add register option at login screen
image

### PR DESCRIPTION
**Pull request by mateuszkula**
I want to merge this change because... it fixes #218 

### Screenshots

"create an account" link has been added:

Zrzut ekranu 2019-05-8 o 13 22 29

I modified Login component so now it can accept prop for setting active tab - user is presented with **Register** tab when clicking "create an account" instead of **Sign in** tab

Zrzut ekranu 2019-05-8 o 13 22 40

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

File: src/components/CheckoutLogin/CheckoutAsGuest.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+
+import { Link } from "react-router-dom";
+import { Button, OverlayTheme, OverlayType } from "..";
+
+const CheckoutAsGuest = ({ overlayContext, checkoutUrl }) => (
```
Comment (piotrgrundas) [inline src/components/CheckoutLogin/CheckoutAsGuest.tsx]: Types

File: src/components/CheckoutLogin/ResetPasswordForm.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import { PasswordResetForm } from "..";
+
+const ResetPasswordForm = ({ onClick }) => (
```
Comment (piotrgrundas) [inline src/components/CheckoutLogin/ResetPasswordForm.tsx]: Ditto

File: src/components/CheckoutLogin/index.tsx [hunk 3]
```diff
@@ -23,71 +20,49 @@ class CheckoutLogin extends React.PureComponent<
...
-                              }
-                            >
-                              Click Here
-                            </span>
-                          </p>
+            <UserContext.Consumer>
+              {({ user }) => {
+                if (user) {
+                  return <Redirect to={checkoutUrl} />;
+                }
```
Comment (piotrgrundas) [inline src/components/CheckoutLogin/index.tsx]: U could introduce hooks here for convenience.

File: src/components/CheckoutLogin/SignInForm.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { LoginForm } from "../";
+
+const SignInForm = ({ onClick }) => (
```
Comment (piotrgrundas) [inline src/components/CheckoutLogin/SignInForm.tsx]: Ditto

File: src/components/CheckoutLogin/ResetPasswordForm.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  <>
+    <h3 className="checkout__header">Registered user</h3>
+    <PasswordResetForm />
+    <div className="login__content__password-reminder">
```
Comment (piotrgrundas) [inline src/components/CheckoutLogin/ResetPasswordForm.tsx]: This `BEM` looks really weird

File: src/components/CheckoutLogin/CheckoutAsGuest.tsx (hunk 1 above)
Comment (mateuszkula) [reply src/components/CheckoutLogin/CheckoutAsGuest.tsx]: Types added

File: src/components/CheckoutLogin/ResetPasswordForm.tsx (hunk 2 above)
Comment (mateuszkula) [reply src/components/CheckoutLogin/ResetPasswordForm.tsx]: Fixed

File: src/components/CheckoutLogin/index.tsx (hunk 3 above)
Comment (mateuszkula) [reply src/components/CheckoutLogin/index.tsx]: Hooks added :)

File: src/components/CheckoutLogin/SignInForm.tsx (hunk 4 above)
Comment (mateuszkula) [reply src/components/CheckoutLogin/SignInForm.tsx]: Added hooks

File: src/components/CheckoutLogin/ResetPasswordForm.tsx (hunk 5 above)
Comment (mateuszkula) [reply src/components/CheckoutLogin/ResetPasswordForm.tsx]: Done

Comment (mateuszkula) [reply src/components/CheckoutLogin/ResetPasswordForm.tsx]: Added generic u-link class which is ussed in few places

File: src/components/CheckoutLogin/index.tsx [hunk 6]
```diff
@@ -1,95 +1,64 @@
...
-                    )}
-                  </div>
+const CheckoutLogin: React.FC<{}> = () => {
+  const [resetPassword, setResetPassword] = useState(false);
+  return (
+    <OverlayContext.Consumer>
+      {overlayContext => {
+        return (
+          <UserContext.Consumer>
+            {({ user }) => {
```
Comment (piotrgrundas) [inline src/components/CheckoutLogin/index.tsx]: There r hooks for contexts as well :)

File: src/components/CheckoutLogin/CheckoutAsGuest.tsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+import { OverlayContextInterface } from "../Overlay";
+
+const CheckoutAsGuest: React.FC<{
+  overlayContext: OverlayContextInterface;
+  checkoutUrl: string;
+}> = ({ overlayContext, checkoutUrl }) => (
+  <div className="checkout-login__guest">
+    <h3 className="checkout__header">Continue as a guest</h3>
+    <p>
+      If you don’t want to register you account at our store don’t worry. You
```
Comment (NyanKiyoshi) [inline src/components/CheckoutLogin/CheckoutAsGuest.tsx]:
```suggestion
      If you don’t wish to register an account, don’t worry. You
```

File: src/components/CheckoutLogin/CheckoutAsGuest.tsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+      can finish your checkout as a guest. You’ll be treated just as good as a
```
Comment (NyanKiyoshi) [inline src/components/CheckoutLogin/CheckoutAsGuest.tsx]: Avoid using passive phrases 👍 

```suggestion
      can checkout as a guest. We care about you just as good as any
```

**Comment by piotrgrundas** [comment]
Please click `Resolve conversation` if u fixed sth.

File: src/components/CheckoutLogin/index.tsx (hunk 6 above)
Comment (mateuszkula) [reply src/components/CheckoutLogin/index.tsx]: Added :)

**Pull request merged**